cardano-node/src/Cardano/Node/ToObjectOrphans.hs

```diff
@@ -0,0 +1,188 @@
+import           Data.Text (Text, pack)
+import           Cardano.BM.Data.Tracer (ToObject (..),
+                                         TracingVerbosity (..),
+                                         Transformable (..),
+                                         emptyObject, mkObject, trStructured)
+
+import           Ouroboros.Consensus.Ledger.Abstract
+import           Ouroboros.Consensus.Util.Condense
+import qualified Ouroboros.Network.AnchoredFragment as AF
+import           Ouroboros.Network.Block
+import qualified Ouroboros.Storage.ChainDB as ChainDB
+import qualified Ouroboros.Storage.LedgerDB.OnDisk as LedgerDB
+
+
+-- | Tracing wrapper which includes current tip in the logs (thus it requires
+-- it from the context).
+--
+-- TODO: this should be moved to `ouroboros-consensus`.  Running in a seprate
+-- STM transaction we risk reporting  wrong tip.
+--
+data WithTip blk a =
+    WithTip
+      (Point blk)
+      -- ^ current tip point
+      a
+      -- ^ data
+
+showWithTip :: Condense (HeaderHash blk)
+            => (a -> String)
+            -> WithTip blk a
+            -> String
+showWithTip customShow (WithTip tip a) = "[" ++ (showTip tip) ++ "] " ++ customShow a
+
+showTip :: Condense (HeaderHash blk)
+        => Point blk
+        -> String
+showTip tip = case pointHash tip of
+    GenesisHash -> "genesis"
+    BlockHash h -> take 7 (condense h)
+
+instance ( Show a
+         , Condense (HeaderHash blk)
+         ) => Show (WithTip blk a) where
+
+    show = showWithTip show
+
+
+
+-- | instances of ToObject
+
+instance (Condense (HeaderHash blk), ProtocolLedgerView blk)
+            => Transformable Text IO (WithTip blk (ChainDB.TraceEvent blk)) where
+    trTransformer = trStructured -- structure required, will call 'toObject'
+instance (Condense (HeaderHash blk), ProtocolLedgerView blk)
+            => ToObject (WithTip blk (ChainDB.TraceEvent blk)) where
+    -- toObject MinimalVerbosity _ = emptyObject -- no output
+    toObject verb (WithTip tip ev) =
+        let evobj = toObject verb ev
+        in
+        if evobj == emptyObject
+        then emptyObject
+        else mkObject [ "kind" .= String "TraceEvent"
+                      , "tip" .= showTip tip
+                      , "event" .= evobj
+                      ]
+
+instance ToObject SlotNo where
+    toObject _verb slot =
+        mkObject [ "kind" .= String "SlotNo"
+                 , "slot" .= toJSON (unSlotNo slot) ]
+
+instance (Condense (HeaderHash blk), ProtocolLedgerView blk)
+    => ToObject (Point blk) where
+    toObject MinimalVerbosity p = toObject NormalVerbosity p
+    toObject verb p =
+        mkObject [ "kind" .= String "Tip"
+                , "tip" .= String (showTip' verb p) ]
+showTip' :: (Condense (HeaderHash blk))
+        => TracingVerbosity -> Point blk -> Text
+showTip' verb tip =
+    let n :: Int = case verb of
+                MinimalVerbosity -> 7
+                NormalVerbosity -> 7
+                MaximalVerbosity -> 99
```

Comment:
Instead of using 99 as a proxy for "the whole hash" (what if the hash length is 128? I also don't like the idea of using `maxBound`), I would do the following:
```haskell
showTip' :: Condense (HeaderHash blk)
         => TracingVerbosity -> Point blk -> Text
showTip' verb tip = case pointHash tip of
    GenesisHash -> "genesis"
    BlockHash h -> pack $ trim $ condense h
  where
    trim = case verb of
      MinimalVerbosity -> take 7
      NormalVerbosity  -> take 7
      MaximalVerbosity -> id
```

Reply:
:+1: